app/src/main/java/com/dcrandroid/fragments/more/PoliteiaFragment.kt

```diff
@@ -0,0 +1,864 @@
+            loadMore -> proposals.size
+            else -> 0
+        }
+
+        val jsonResult = multiWallet.politeia.getProposals(Constants.PROPOSALS_ABANDONED, offset, limit, newestProposalsFirst)
+
+        // Check if the result object from the json response is null
+        val resultObject = JSONObject(jsonResult).get("result")
+        val resultObjectString = resultObject.toString()
+        val abandonedProposalList = if (resultObjectString == "null") {
+            gson.fromJson("[]", Array<Proposal>::class.java)
+        } else {
+            val resultArray = JSONObject(jsonResult).getJSONArray("result")
+            val resultArrayString = resultArray.toString()
+            gson.fromJson(resultArrayString, Array<Proposal>::class.java)
+        }
+
+        initialLoadingDone.set(true)
+
+        if (abandonedProposalList == null) {
+            loadedAll = true
+            loading.set(false)
+            hideLoadingView()
+
+            if (!loadMore) {
+                proposals.clear()
+            }
+            return@launch
+        }
+
+        if (abandonedProposalList.size < limit) {
+            loadedAll = true
+        }
+
+        if (loadMore) {
+            val positionStart = proposals.size
+            proposals.addAll(abandonedProposalList)
+            withContext(Dispatchers.Main) {
+                proposalAdapter?.notifyItemRangeInserted(positionStart, abandonedProposalList.size)
+
+                // notify previous last item to remove bottom margin
+                proposalAdapter?.notifyItemChanged(positionStart - 1)
+            }
+
+        } else {
+            proposals.let {
+                it.clear()
+                it.addAll(abandonedProposalList)
+            }
+
+            checkEmptyProposalList("abandoned")
+
+            withContext(Dispatchers.Main) {
+                proposalAdapter?.notifyDataSetChanged()
+            }
+        }
+
+        loading.set(false)
+        hideLoadingView()
+
+        swipe_refresh_layout.isRefreshing = false
+    }
+
+    private fun checkEmptyProposalList(status: String) {
+        activity!!.runOnUiThread {
+            if (proposals.size > 0) {
+                swipe_refresh_layout?.show()
+            } else {
+                swipe_refresh_layout?.hide()
+                empty_list.text = String.format(Locale.getDefault(), "No %s proposals", status)
+            }
+        }
+    }
+
+    private fun refreshAvailableProposalCategories() = GlobalScope.launch(Dispatchers.Default) {
+        availableProposalTypes.clear()
+
+        val preCount = multiWallet.politeia.count(Constants.PROPOSALS_PRE)
+        val activeCount = multiWallet.politeia.count(Constants.PROPOSALS_ACTIVE)
+        val approvedCount = multiWallet.politeia.count(Constants.PROPOSALS_APPROVED)
+        val rejectedCount = multiWallet.politeia.count(Constants.PROPOSALS_REJECTED)
+        val abandonedCount = multiWallet.politeia.count(Constants.PROPOSALS_ABANDONED)
```

Comment:
Use dcrlibwallet constants
```suggestion
        val preCount = multiWallet.politeia.count(Dcrlibwallet.ProposalCategoryPre)
        val activeCount = multiWallet.politeia.count(Dcrlibwallet.ProposalCategoryActive)
        val approvedCount = multiWallet.politeia.count(Dcrlibwallet.ProposalCategoryApproved)
        val rejectedCount = multiWallet.politeia.count(Dcrlibwallet.ProposalCategoryRejected)
        val abandonedCount = multiWallet.politeia.count(Dcrlibwallet.ProposalCategoryAbandoned)
```